test/macro.jl

```diff
@@ -0,0 +1,43 @@
+using Reactive
+using FactCheck
+using Compat
+
+a = Input(2)
+b = @lift a^2
+facts("@lift") do
+
+    context("@lift input expressions") do
+
+        t1 = @lift (a,)
+        t2 = @lift (a, b)
+        l1 = @lift [a]
+        l2 = @lift [a, b^2]
+        c1 = @lift {a}
+
+        push!(a, 3)
+
+        @fact t1.value => (a.value,)
+        @fact t2.value => (a.value, b.value)
+        @fact l1.value => [a.value]
+        @fact l2.value => [a.value, b.value^2]
+        @fact c1.value =>  {a.value}
+    end
+    context("@lift basics") do
+        @fact value(a)^2 => value(b)
+
+        push!(a, 3)
+
+        @fact a.value^2 => b.value
+    end
+
+    # test use in a function
+    context("@lift inside a function") do
+        k = 3
+        f(a,b) = @lift a + b + 1 + k
```

Comment:
@tshort This is still broken,

First, it says k is not defined. However, if I copy this to the REPL, the problem goes away. But then it screams a and b are not defined unless I have a and b in the outer scope, which does not feel correct...

I reverted some changes irrelevant to this feature from your PR, please take a look at the macros.jl in this branch. Was this case working when you tried??